src/backend/distributed/operations/auto_shard_split.c

```diff
@@ -0,0 +1,141 @@
+#include "distributed/background_jobs.h"
+#include "distributed/citus_safe_lib.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/colocation_utils.h"
+#include "distributed/connection_management.h"
+#include "distributed/enterprise.h"
+#include "distributed/hash_helpers.h"
+#include "distributed/listutils.h"
+#include "distributed/lock_graph.h"
+#include "distributed/coordinator_protocol.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/metadata_utility.h"
+#include "distributed/multi_logical_replication.h"
+#include "distributed/multi_progress.h"
+#include "distributed/multi_server_executor.h"
+#include "distributed/pg_dist_rebalance_strategy.h"
+#include "distributed/pg_dist_shard.h"
+#include "distributed/reference_table_utils.h"
+#include "distributed/remote_commands.h"
+#include "distributed/resource_lock.h"
+#include "distributed/shard_rebalancer.h"
+#include "distributed/shard_cleaner.h"
+#include "distributed/shard_transfer.h"
+#include "distributed/tuplestore.h"
+#include "distributed/utils/array_type.h"
+#include "distributed/worker_protocol.h"
+#include "nodes/pg_list.h"
+#include "funcapi.h"
+#include "miscadmin.h"
+#include "postmaster/postmaster.h"
+#include "storage/lmgr.h"
+#include "utils/builtins.h"
+#include "utils/fmgroids.h"
+#include "utils/json.h"
+#include "utils/lsyscache.h"
+#include "utils/memutils.h"
+#include "utils/pg_lsn.h"
+#include "utils/syscache.h"
+#include "common/hashfn.h"
+#include "utils/varlena.h"
+#include "utils/guc_tables.h"
+#include "distributed/commands/utility_hook.h"
+
+#define MAX_SHARD_SIZE  1
+PG_FUNCTION_INFO_V1(citus_auto_shard_split_start); 
+
+void Citus_Table_List(void)
+{
+	Assert(CitusHasBeenLoaded() && CheckCitusVersion(WARNING));
+
+	/* first, we need to iterate over pg_dist_partition */
+	List *citusTableIdList = CitusTableTypeIdList(DISTRIBUTED_TABLE);
+
+	Oid relationId = InvalidOid;
+	foreach_oid(relationId, citusTableIdList)
+	{
+		CitusTableCacheEntry *cacheEntry = GetCitusTableCacheEntry(relationId);
+        Oid tempRelationId = cacheEntry->relationId;
+        List *tuplesFromEachDistributedTable =LookupDistShardTuples(tempRelationId);
+        HeapTuple tempTuple;
+
+        ListCell* cell;
+        foreach(cell, tuplesFromEachDistributedTable) {
+            tempTuple = lfirst(cell);
+            Form_pg_dist_shard distShard = (Form_pg_dist_shard)GETSTRUCT(tempTuple);
+            ShardInterval *shardInterval = LoadShardInterval(distShard->shardid);
+            int32 shardMinValue = DatumGetInt32(shardInterval->minValue);
+            int32 shardMaxValue = DatumGetInt32(shardInterval->maxValue);
+
+            ereport(LOG, (errmsg("Shard ID: %d,ShardMinValue: %d, ShardMaxValue: %d", distShard->shardid,shardMinValue,shardMaxValue)));
+        }
+        
+       
+        
+    }
+}
+
+void Citus_get_Data_by_view(){
+    bool missingOk = false;
+    int connectionFlags = 0;
+    WorkerNode *workerNode = FindNodeWithNodeId(1, missingOk);
```

Comment:
there may not be a node 1. In general, its better to get this information using internal C functions.